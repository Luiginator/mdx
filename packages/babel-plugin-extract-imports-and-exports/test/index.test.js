const fixtures = require('./fixtures/')

const babel = require('@babel/core')
const BabelPluginExtractImportNames = require('..')

const transform = str => {
  const plugin = new BabelPluginExtractImportNames()

  babel.transform(str, {
    configFile: false,
    plugins: ['@babel/plugin-syntax-jsx', plugin.plugin]
  })

  return plugin.state
}

fixtures.forEach(fixture => {
  it(fixture.description, () => {
    expect(transform(fixture.mdx)).toMatchSnapshot(fixture.result)
  })
})
