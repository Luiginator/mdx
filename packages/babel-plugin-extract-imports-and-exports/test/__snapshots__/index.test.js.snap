// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handles an import that is later exported 1`] = `
Object {
  "nodes": Array [
    Object {
      "start": 0,
      "type": "import",
    },
    Object {
      "default": true,
      "start": 28,
      "type": "export",
    },
  ],
}
`;

exports[`Handles basic usecase of default export 1`] = `
Object {
  "nodes": Array [
    Object {
      "default": true,
      "start": 0,
      "type": "export",
    },
  ],
}
`;

exports[`Handles const exports 1`] = `
Object {
  "nodes": Array [
    Object {
      "start": 0,
      "type": "export",
    },
  ],
}
`;

exports[`Handles multiline default exports 1`] = `
Object {
  "nodes": Array [
    Object {
      "default": true,
      "start": 0,
      "type": "export",
    },
  ],
}
`;

exports[`Handles multiline exports 1`] = `
Object {
  "nodes": Array [
    Object {
      "start": 0,
      "type": "export",
    },
  ],
}
`;

exports[`Separates import from the default export 1`] = `
Object {
  "nodes": Array [
    Object {
      "start": 0,
      "type": "import",
    },
    Object {
      "default": true,
      "start": 24,
      "type": "export",
    },
  ],
}
`;
